File: 1603-Design-Parking-System.cpp
class ParkingSystem {
    int slots[3];

   public:
    ParkingSystem(int big, int medium, int small) {
        slots[0] = big;
        slots[1] = medium;
        slots[2] = small;
    }

    bool addCar(int carType) {
        // convert carType begin with 0
        carType--;
        if (slots[carType] <= 0) return false;
        slots[carType]--;
        return true;
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */

File: 1603-Design-Parking-System.go
type ParkingSystem struct {
    int slots[4]
}


func Constructor(big int, medium int, small int) ParkingSystem {
    return ParkingSystem{{0, big, medium, small}}
}


func (this *ParkingSystem) AddCar(carType int) bool {
    if slots[carType] > 0 {
        slots[carType]--
        return true
    }
    return false
}


/**
 * Your ParkingSystem object will be instantiated and called as such:
 * obj := Constructor(big, medium, small);
 * param_1 := obj.AddCar(carType);
 */